src/main/steps/respondent/consent-to-application/ConsentMapper.ts

```diff
@@ -1,59 +1,42 @@
 import { CaseWithId } from '../../../app/case/case';
 import { Consent, Respondent, YesOrNo } from '../../../app/case/definition';
-import { fromApiDate } from '../../../app/case/from-api-format';
-import { toApiDate } from '../../../app/case/to-api-format';
-import type { AppRequest } from '../../../app/controller/AppRequest';
-
-export const setConsentDetails = (respondent: Respondent, req: AppRequest): Respondent => {
-  let consentFromResponsent: Consent;
-  let reasonForNotConsenting = '';
-  let courtOrderDetails = '';
-  if (req.session.userCase.doYouConsent === YesOrNo.NO) {
-    reasonForNotConsenting = req.session.userCase.reasonForNotConsenting!;
-  }
-  if (req.session.userCase.courtPermission === YesOrNo.YES) {
-    courtOrderDetails = req.session.userCase.courtOrderDetails!;
+
+export const prepareRequest = (req: CaseWithId): Consent => {
+  const { doYouConsent, courtPermission, reasonForNotConsenting, courtOrderDetails } = req;
+
+  const request: Consent = {};
+
+  Object.assign(request, {
+    consentToTheApplication: doYouConsent,
+    permissionFromCourt: courtPermission,
+    noConsentReason: reasonForNotConsenting,
+    courtOrderDetails,
+  });
+
+  // data cleanup
+  if (doYouConsent === YesOrNo.YES) {
+    delete request.noConsentReason;
   }
-  if (respondent?.value?.response && respondent?.value?.response?.consent) {
-    consentFromResponsent = respondent?.value?.response?.consent;
-    consentFromResponsent.consentToTheApplication = req.session.userCase.doYouConsent;
-    consentFromResponsent.noConsentReason = reasonForNotConsenting;
-    consentFromResponsent.applicationReceivedDate = toApiDate(req.session.userCase.applicationReceivedDate);
-    consentFromResponsent.permissionFromCourt = req.session.userCase.courtPermission;
-    consentFromResponsent.courtOrderDetails = courtOrderDetails;
-    respondent.value.response.consent = consentFromResponsent;
-  } else {
-    respondent.value.response = {
-      consent: {
-        consentToTheApplication: req.session.userCase.doYouConsent,
-        noConsentReason: reasonForNotConsenting,
-        applicationReceivedDate: toApiDate(req.session.userCase.applicationReceivedDate),
-        permissionFromCourt: req.session.userCase.courtPermission,
-        courtOrderDetails,
-      },
-    };
+
+  if (courtPermission === YesOrNo.NO) {
+    delete request.courtOrderDetails;
   }
-  return respondent;
+
+  return request;
 };
 
-export const getConsentDetails = (respondent: Respondent, req: AppRequest): Partial<CaseWithId> => {
-  if (respondent?.value?.response?.consent?.consentToTheApplication === YesOrNo.NO) {
```

Comment:
Why is the existing data transformation conditions removed? Where have you handled all these in the mapDetails?

Reply:
The conditions were removed, because the code reaches the `mapConsentToApplicationDetails` method, they are already cleaned up and there's no need to for a further logical comparison.